destination.go

```diff
@@ -214,7 +226,10 @@ func (a *destinationPluginAdapter) waitForAcks(ctx context.Context) error {
 func (a *destinationPluginAdapter) Stop(ctx context.Context, req cpluginv1.DestinationStopRequest) (cpluginv1.DestinationStopResponse, error) {
 	// flush cached records
 	err := a.impl.Flush(ctx)
-	if err != nil {
+	if err != nil &&
+		// only propagate error if we are sure the plugin is writing records
+		// asynchronously or if it's some other error than ErrUnimplemented
+		(a.isAsync || !errors.Is(err, ErrUnimplemented)) {
```

Comment:
Relevant code snippet for making `Destination.Flush` optional for plugins that synchronously write records.